### ISSUE TITLE
Disable generating bin files

### ISSUE THREAD

**Issue by MikeLund** [composer/composer#5538]
Is there any way to make it so composer will not create the vendor/bin/ items?

I know I can set COMPOSER_BIN_DIR to somewhere else, or delete the folder after it's made... but is there any way to fully disable their creation globally?

My project requires laravel/framework, which then requires classpreloader/classpreloader, which then requires nikic/php-parser. Tada, php-parser's composer.json contains ``"bin": ["bin/php-parse"]``. So I get vendor/bin/php-parse created when I run ``composer install`` on a new system.

Of course it's not a big deal whatsoever because it's simple to manually delete stuff afterwards, but it feels weird if there isn't an option to disable this feature.

tl;dr: so, is it currently possible to *disable the creation of* the vendor binary symlinks, other than removing the "bin" directives from tons of dependencies? Thanks!

**Comment by alcohol** [comment]
To my knowledge, there is currently no functionality to explicitly disable this.

**Comment by hkdobrev** [comment]
If there is a configuration option on the project for that, people would probably need to whitelist binaries. Or another way is to create symlinks to binaries only for root dependencies, but not for binaries from dependencies on dependencies.

**Comment by hkdobrev** [comment]
@MikeLund At the moment you could try with setting `COMPOSER_BIN_DIR` to `/dev/null` on Unix-like systems. Haven't tried it though.

**Comment by MikeLund** [comment]
@hkdobrev I set it to ~/.Trash and it works fine. Isn't really an issue or anything, just really was seeing if there was some alternative way that is "cleaner", but guess not.

So it probably isn't worth any dev's time to add this feature!

**Comment by Seldaek** [comment]
Yeah I don't think this is worth implementing really, a few symlinks don't hurt, and by default it's hidden inside the vendor dir anyway.
